### PR TITLE
Sync common lint scripts and configs

### PR DESCRIPTION
**Pull request by at-wat**
This PR sets up mechanism to sync common scripts/configs placed in multiple repositories.
Adding a version tag to this repository triggers GitHub Action which creates PRs syncing the files to each repository.

### TODO
- [x] create a bot account @pionbot
- [x] register a personal access token of the bot as `secrets.GITHUB_TOKEN_REPO`
- [x] add `don't edit directly` comment to each file
- [x] add usage to README

### Related
- https://github.com/pion/webrtc/pull/960#issuecomment-570787506

### CODE REVIEW

**Comment by at-wat** [comment]
@Sean-Der it's almost ready to do https://github.com/pion/webrtc/pull/960#issuecomment-569396681.

File: root/.github/assert-contributors.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+# Unshallow the repo, this check doesn't work with this enabled
+# https://github.com/travis-ci/travis-ci/issues/3412
+if [ -f $(git rev-parse --git-dir)/shallow ]; then
+	git fetch --unshallow || true
+fi
+
+SCRIPT_PATH=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
+
+EXCLUDED_CONTIBUTORS=('John R. Bradley' 'renovate[bot]' 'Renovate Bot')
```
Comment (at-wat) [inline root/.github/assert-contributors.sh]: should have `Pion Bot` also

**Comment by at-wat** [comment]
@Sean-Der @maxhawkins I have copied these lint and hook scripts from `pion/webrtc` and added you to contributor list in README. Please take a quick look!

**Comment by Sean-Der** [comment]
This is really amazing @at-wat thank you so much for taking this on. This has been something I have wanted for two years and just didn't put the effort forward.

I can't oversell what an improvement this is going to bring to the quality of those little repos we aren't paying attention too. This will help up keep the `golangci-lint` up to date and other great things like that.

Thank you so much

**Pull request merged**